### PR TITLE
Added the ability to specify the S3 credentials explicitly for the RDS & S3 automated setup

### LINKED ISSUE #99: Add the ability to specify the S3 credentials explicitly for the RDS & S3 automated setup
**Is your feature request related to a problem? Please describe.**
Users might want to use the credentials of a specific account for managing s3 that is different than the credentials of the user running the script. 

**Describe the solution you'd like**
Add an optional parameter to the script that allows to explicitly use a specific access key and secret key.

### PR DESCRIPTION
**Pull request by AlexandreBrown**
**Which issue is resolved by this Pull Request:**
Resolves #99   
  
Must be merged after https://github.com/awslabs/kubeflow-manifests/pull/105

**Checklist:**
- [x] Unit tests pass:
  **Make sure you have installed kustomize == 3.2.1**
    1. `make generate-changed-only`
    2. `make test`
- [x] end-to-end test from fresh cluster (main branch + manifest v1.4.1)

### CODE REVIEW

File: docs/deployment/rds-s3/README.md [hunk 1]
```diff
@@ -53,7 +53,38 @@ eksctl create cluster \
...
+The script takes care of creating the S3 bucket, creating the S3 secrets using the secrets manager, setting up the RDS database and creating the RDS secret using the secrets manager. It also edits the required configuration files for Kubeflow pipeline to be properly configured for the RDS database during Kubeflow installation.  
+The script also handles cases where the resources already exist in which case it will simply skips the step.
+  
+Note : The script will **not** delete any resource therefore if a secret already exist or if a database with the same name already exist or if the S3 bucket already exists for instance, it will skip those steps. This was done in order to prevent unwanted results such as accidental deletion. For instance, if a database with the same name already exists, the script will skip the database creation setup. If it's expected in your scenario, then perhaps this is fine for you, if you simply forgot to change the database name used for creation then this gives you the chance to retry the script with the proper value. See `python auto-rds-s3-setup.py --help` for the list of parameters as well as their default values.
```
Comment (goswamig) [inline docs/deployment/rds-s3/README.md:68]: If resource already exist, I believe your script will use it. In that case we should call it out here.

```
If resources (bucket, RDS, secrets) already exists, the script will use it as it is.
```

File: docs/deployment/rds-s3/auto-rds-s3-setup.py [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+    print("Secrets IAM service account created!")
+
+
+def setup_secrets_provider():
+    print("Installing secrets provider...")
+    install_secrets_store_csi_driver()
+    print("Secrets provider install done!")
+
+
+def install_secrets_store_csi_driver():
```
Comment (goswamig) [inline docs/deployment/rds-s3/auto-rds-s3-setup.py:495]: Are these the only files necessary for driver installations. ? If you're following any docs for this, please add it here in comment.

**Review by goswamig — COMMENTED** [review]
Great work!!! 

Thanks for sending this PR , left a minor comment.

File: docs/deployment/rds-s3/auto-rds-s3-setup.py (hunk 2 above)
Comment (AlexandreBrown) [reply docs/deployment/rds-s3/auto-rds-s3-setup.py:495]: Yes these come from the manual steps (see step #4) from https://github.com/awslabs/kubeflow-manifests/tree/v1.3-branch/distributions/aws/examples/rds-s3#22-option-2-manual-setup

File: docs/deployment/rds-s3/README.md (hunk 1 above)
Comment (AlexandreBrown) [reply docs/deployment/rds-s3/README.md]: Great suggestion, I reworked the doc to follow your suggestion.

**Review by akartsky — APPROVED** [review]
LGTM thanks 

Can we also add arguments for RDS credentials too just to be consistent 
This could be a new PR

**Pull request merged**